### DISCUSSION TITLE
Using container checkout doesn't checkout hidden files

### DISCUSSION THREAD

**Discussion by dingo-d** [community/community#26978] [Actions]
<p>I have the following job set up</p>
<pre><code>tests:
        name: Integration tests
        runs-on: ubuntu-latest

        container:
            image: wordpress:5.4.1-php7.2
            env:
                WORDPRESS_DB_HOST: mysql
                WORDPRESS_DB_USER: admin
                WORDPRESS_DB_PASSWORD: ""
                WORDPRESS_DB_NAME: wordpresstest
            ports:
                - 8080:80
            volumes:
                - wordpress:/var/www/html/:rw
            options: --cpus 2

        services:
            mysql:
                image: mysql:5.7
                env:
                    MYSQL_ALLOW_EMPTY_PASSWORD: yes
                    MYSQL_DATABASE: wordpresstest
                ports:
                    - 3306
                options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3

        steps:
            - name: Checkout
              uses: actions/checkout@v2

            - name: PHP setup
              uses: shivammathur/setup-php@v2
              with:
                php-version: 7.2
                coverage: pcov

            - name: Set Node.js version
              uses: actions/setup-node@v1
              with:
                node-version: 14

            - name: Copy .env.testing
              run: yes | cp .env.actions.testing .env.testing
</code></pre>
<p>it’s not fully complete because I have two issues here (and I’m probably not understanding things right).</p>
<p>When I get to the last step it fails because none of my hidden files (except .gitattributes and .gitignore) are present in the checkout phase. But when I had wordpress as a service and not a container, everything was checked out correctly. </p>
<p>Another thing that confuses me is, if I’m in a container (in my case WordPress one), shouldn’t I see all the WordPress files when I do <code>ls -all</code>?</p>
<p>Or are the steps executed on the GH host machine, and the container is started, but the only way to access it would be to exec into it?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/dingo-d">@dingo-d</a> ,</p>
<p>Please use ‘v1’ for checkout action. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f7aabeca805e363baaeac62bfc62f2e4b6deac9c.png" data-download-href="/uploads/short-url/zkXKPg91wF3KnLRInBnCRstre2M.png?dl=1" title="checkoutcontainer.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181101033-7a2cbf68-4cca-41ce-8792-cb3f731e2fdb.png" alt="checkoutcontainer.png" data-base62-sha1="zkXKPg91wF3KnLRInBnCRstre2M" width="690" height="346" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f7aabeca805e363baaeac62bfc62f2e4b6deac9c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">checkoutcontainer.png</span><span class="informations">727×365 19.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When container is defined, your job steps are executed on the docker container, which is different with services(steps executed on github runner).</p>
<p>It seems checkout v2 action excludes the files which start with dot, it’s recommended to raise a query on the <a href="https://github.com/actions/checkout/issues" rel="nofollow noopener">list</a> for confirmation.</p>
<p>Thanks.</p>

**Comment by dingo-d** [comment]
<p>Thanks! I’ve added an issue on the repo, in the meantime I’ll try with v1.</p>
